---
title: "Gene List Enrichment Heatmaps"
author: "Sara Gosline"
date: "2018"
output: html_document
params:
  samp.mat: NA
  cell.annotations: NA
  seqData: TRIE
---

###Properly formatted data
```{r,echo=FALSE} 
knitr::opts_chunk$set(echo = TRUE)

samp.mat <- params$samp.mat
cell.annotations <- params$cell.annotations
seqData<-params$seqData
```

## R Markdown

We may also have cell-specific annotations, These two are experiment-specific and should be formatted in a way that can be better visualized. These are passed in as `cell.annotations`:
```{r see-annotations}
head(cell.annotations)
```
### Single Cell Enrichment
We have also collated a set of gene lists that are known to be expressed in single cell populations.

```{r Get Genes,results='asis'}
gene.lists<-c('mcpcounter','cibersort','LyonsEtAl','Wallet','SchelkerEtAl')
require(pheatmap)
 suppressMessages(library(GSVA))
# 
 for(gl in gene.lists){
   g.list<- singleCellSeq::getGeneList(gl)
  # g.list<-lapply(cell.annotations%>%split(.$Cell),function(x) x$Gene)
   g.res<-gsva(as.matrix(samp.mat),g.list,method='ssgsea',rnaseq=seqData,verbose=FALSE)
   colnames(g.res)<-rownames(cell.annotations)
   cat('\n\n')  
   cat(paste("#### ssGSEA of",gl))
   suppressWarnings(pheatmap(t(g.res),cellwidth=10,clustering_distance_rows='correlation',clustering_distance_cols='correlation',clustering_method='ward.D2',annotation_row=cell.annotations[colnames(g.res),],show_rownames=F))
   cat('\n\n') 
 }
```
  
