---
title: "scRNA-Seq Annotation"
author: "Sara Gosline"
date: "July 3, 2018"
output: html_document
editor_options: 
  chunk_output_type: console

params:
  samp.mat: NA
  gene.list: NA
---

## scRNA-Seq Immune Cell Annotation

The goal of this document is to demonstrate the basic steps/tools available to pull scRNA-seq data from [Synapse](http://www.synapse.org/scRNASeq) and use available annotations to visualize the results. 

###Properly formatted data
```{r,echo=FALSE} 
samp.mat <- params$samp.mat
gene.list <-params$gene.list

```

All data should be in a matrix form where the rows represent genes and the columns represent samples. Please see the [scRNA-Seq Instructions](https://www.synapse.org/#!Synapse:syn11857287/wiki/518624) for more details. These files should be passed in as `samp.mat`.

```{r see-sample}
samp.mat[1:10,1:10]
```

Now that the data are processed we can begin to cluster and annotate the cells based on gene expression levels.

###Grouping cells via clustering
The first step of analysis is to group the cells using clustering. Here we implement two approaches but can add more. 

#### Plot cells in two dimensions

There are multiple ways to cluster the cells, with many more methods in development. 

###### Principle Component Analysis

```{r Compare clustering}
#use example function
res.s <-singleCellSeq::dataMatrixToCluster.seurat(samp.mat)

```


#### Assign cells to cluster
With these clusters we then need to group assign cells into groups. 

```{r Cluster Cells}
clust.sres<-Seurat::FindClusters(res.s)
```

#### Visualize Clusters


With Seurat just makes this much easier!
```{R Seurat Clusters}
Seurat::PCAPlot(clust.sres)
Seurat::TSNEPlot(clust.sres)
```

Now we can see the clusters and evaluate the gene lists of interest.
```{r genes}
genes<-singleCellSeq::getGeneList(gene.list)

by.cell=lapply(as.character(unique(genes$Cell)),function(x) subset(genes,Cell==x)$Gene)
names(by.cell)<-as.character(unique(genes$Cell))

#now create file paths
paths <- paste0(names(by.cell), ".png")
names(paths)<-names(by.cell)

try(dir.create('plots'))
all.plots<-lapply(names(by.cell),function(gc){
  gs<-by.cell[[gc]]
  gs<-gs[which(gs%in%rownames(clust.sres@data))]
  png(paste0("plots/",paths[[gc]]))
  FeaturePlot(object = clust.sres, features.plot = gs, cols.use = c("grey", "blue"), reduction.use = "tsne",do.return=TRUE)
  #ggsave(paths[[gc]],path='plots',dpi=300)
  dev.off()
  })

names(all.plots)<-names(by.cell)



#purrr::pwalk(list(paths, all.plots),ggplot2::ggsave, path = "plots", dpi = 300)

cell_plots <- paste0(list.files("plots", full.names = TRUE))
names(cell_plots) <- stringr::str_replace_all(cell_plots, 
                                      c("\\.png" = "", 
                                        "plots/" = ""))

library(bsselectR)
bsselect(cell_plots, type = "img",live_search = TRUE, show_tick = TRUE)

```

